### PR TITLE
[ADD][9.0] purchase_analytic_distribution module

### PR DESCRIPTION
**Pull request by cubells**
New module to use as alternative to old purchase_analytic_plans module.

- Update README
- Update views
- Update file structure
- Add tests

Depends on
- [x] #90

cc @Tecnativa

### CODE REVIEW

File: purchase_analytic_distribution/i18n/purchase_analytic_plans.pot [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Translation of Odoo Server.
```
Comment (pedrobaeza) [inline purchase_analytic_distribution/i18n/purchase_analytic_plans.pot]: Remove this file

File: purchase_analytic_distribution/README.rst [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+help us smashing it by providing a detailed and welcomed feedback.
+
+Credits
+=======
+
+Contributors
+------------
+
+* Odoo S.A.
+* Vicent Cubells <antonio.espinosa@tecnativa.com>
```
Comment (pedrobaeza) [inline purchase_analytic_distribution/README.rst]: Put correctly your address

File: purchase_analytic_distribution/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+#  Copyright (C) 2004-2010 Tiny SPRL (<http://tiny.be>).
+#  Copyright 2017 Tecnativa - Vicent Cubells
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Purchase Analytic Plans",
+    "version": "9.0.1.0.0",
+    "category": "Purchase Management",
+    "author": "Odoo SA, "
```
Comment (pedrobaeza) [inline purchase_analytic_distribution/__openerp__.py]: Are you sure Odoo puts their authorship this way? Check default value of author key in code, which is the one they use.

File: purchase_analytic_distribution/models/purchase.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+from openerp import fields, models
+
+
+class PurchaseOrderLine(models.Model):
+    _name = 'purchase.order.line'
+    _inherit = 'purchase.order.line'
+
+    analytic_distribution_id = fields.Many2one(
+        comodel_name='account.analytic.distribution',
+        string='Analytic Distribution',
```
Comment (pedrobaeza) [inline purchase_analytic_distribution/models/purchase.py]: Put `oldname='analytics_id'` for automatic migration.

File: purchase_analytic_distribution/views/purchase_view.xml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        </field>
+    </record>
+    <record id="view_purchase_order_form_editable_list" model="ir.ui.view">
+        <field name="name">purchase.order.form.editable.list</field>
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.purchase_order_form"/>
+        <field name="groups_id" eval="[(6, 0, [ref('product.group_uom'), ref('purchase.group_analytic_accounting')])]"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='order_line']/tree" position="attributes">
+                <attribute name="editable"/>
```
Comment (pedrobaeza) [inline purchase_analytic_distribution/views/purchase_view.xml]: Why removing the editable attribute? You can put the analytic distribution in the tree itself.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@luismontalba please review this one

File: purchase_analytic_distribution/README.rst [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+=====
+
+After creating an analytic distribution, create a sale order and you can add
+an analytic distribution to every sale order line, to distribute amount line
+to several analytic accounts.
+
+When you confirm sale order and create invoice, analytic distribution are
+propagated to the new invoice. Multiple analytic lines are created when the
+invoice or the entries are confirmed.
+
```
Comment (luismontalba) [inline purchase_analytic_distribution/README.rst:24]: Purchase order instead off sale order?

**Comment by luismontalba** [comment]
Analytic entries are not created when confirming invoice from a purchase order with analytic distribution in order lines, is it related to this module as the readme explains?

**Comment by pedrobaeza** [comment]
This module only passes the analytic distribution from the purchase order to the invoice.

**Review by luismontalba — APPROVED** [review]
Functionality tested on runbot

**Pull request merged**